pyiceberg/table/__init__.py

```diff
@@ -340,6 +340,34 @@ def set_properties(self, properties: Properties = EMPTY_DICT, **kwargs: Any) ->
         updates = properties or kwargs
         return self._apply((SetPropertiesUpdate(updates=updates),))
 
+    def set_ref_snapshot(
+        self,
+        snapshot_id: int,
+        parent_snapshot_id: Optional[int],
+        ref_name: str,
+        type: str,
+        max_age_ref_ms: Optional[int] = None,
+        max_snapshot_age_ms: Optional[int] = None,
+        min_snapshots_to_keep: Optional[int] = None,
+    ) -> Transaction:
+        """Update a ref to a snapshot.
+
+        Returns:
+            The transaction with the set-snapshot-ref staged
+        """
+        updates = SetSnapshotRefUpdate(
+            snapshot_id=snapshot_id,
+            parent_snapshot_id=parent_snapshot_id,
+            ref_name=ref_name,
+            type=type,
+            max_age_ref_ms=max_age_ref_ms,
+            max_snapshot_age_ms=max_snapshot_age_ms,
+            min_snapshots_to_keep=min_snapshots_to_keep,
+        )
+
+        requirements = AssertRefSnapshotId(snapshot_id=parent_snapshot_id, ref="main")
```

Comment:
Could we add [AssertTableUUID](https://github.com/apache/iceberg-python/blob/main/pyiceberg/table/__init__.py#L1010) table requirement as well?

Reply:
Done

Reply:
@syun64 Any specific requirement to add `AssertTableUUID`? This will cause more potential commit conflicts than necessary.

Reply:
Wouldn't we always want AssertTableUUID for this operation? When we apply manage snapshots on a specific table, we always intend to add a snapshot ID or update the snapshot ID of a ref branch (main, etc) based on the specific instance of the table the client is aware of (which is tracked by the table UUID).

I think adding AssertTableUUID will prevent the operation from succeeding when there are concurrent table replace/create operations that could put the table in a bad state -> https://github.com/apache/iceberg/pull/4322

I'm actually curious why this requirement is missing in the Java implementation. Is it possible that it's being injected for any `SIMPLE` TableUpdate types, including snapshot updates?

[UpdateRequirements.java: forUpdateTable](https://github.com/apache/iceberg/blob/c7de6cb345995cb47312edbef6edae2f17fb8aba/core/src/main/java/org/apache/iceberg/UpdateRequirements.java#L50)

[RESTTableOperations.java: SIMPLE type commit](https://github.com/apache/iceberg/blob/c7de6cb345995cb47312edbef6edae2f17fb8aba/core/src/main/java/org/apache/iceberg/rest/RESTTableOperations.java#L137)

Reply:
Hi folks, based on my findings from the Java code above I put up a PR as a followup to this discussion. Maybe we can scope out this TableRequirement from ManageSnapshots/set_ref_update for now https://github.com/apache/iceberg-python/pull/804

Reply:
We just merged in #804 
@chinmay-bhat shall we remove AssertTableUUID from this PR, as it will be automatically attached to all table updates on commit?

Reply:
Removed AssertTableUUID! Thanks for the followup PR @syun64!